packages/hdwallet-provider/src/index.ts

```diff
@@ -215,18 +225,94 @@ class HDWalletProvider {
     });
   }
 
-  public send(
+  private initialize(): Promise<void> {
+    return new Promise(resolve => {
+      this.sendAsync({
```

Comment:
If you call `this.engine.sendAsync` directly here will that obviate the need for the `this.initialized` check in `this.sendAsync`?

Reply:
Why? We still need to wait for the chain id to be fetched.

Reply:
Oh, I see what you mean. This is only for the call to get the chain id. I think that allows us to get rid of the if/then? TypeScript seems to think so!